From 56195ea08d49efa7362b74abc99cb5ec64f58b70 Mon Sep 17 00:00:00 2001
From: Srinivasa Reddy Vundela <vsr@cloudera.com>
Date: Fri, 27 May 2016 13:10:01 -0700
Subject: [PATCH 205/517] Revert "CLOUDERA-BUILD CDH-27093 Checking for the
 required hadoop libraries before starting the
 spark"

This reverts commit 165238c8629942b8c199737bb8bec8ff3bc2c5d0.
---
 bin/spark-class |   16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/bin/spark-class b/bin/spark-class
index fe6ff4a..5d964ba 100755
--- a/bin/spark-class
+++ b/bin/spark-class
@@ -80,24 +80,8 @@ fi
 # The launcher library will print arguments separated by a NULL character, to allow arguments with
 # characters that would be otherwise interpreted by the shell. Read that in a while loop, populating
 # an array that will be used to exec the final command.
-CP_INDEX=-1
-INDEX=0
 CMD=()
 while IFS= read -d '' -r ARG; do
-  if [ "$ARG" = "-cp" ]; then
-    CP_INDEX=$((INDEX+1))
-  fi
   CMD+=("$ARG")
-  (( INDEX++ ))
 done < <("$RUNNER" -cp "$LAUNCH_CLASSPATH" org.apache.spark.launcher.Main "$@")
-
-# Check for the required libraries before starting spark. Exit if any of the required libraries
-# are missing.
-if [[ ! "${CMD[$CP_INDEX]}" =~ "hadoop-common" ]]; then
-  echo "Required Hadoop libraries are missing in classpath." 1>&2
-  echo "This could happen if gateway roles are not setup on the node or configuration files are missing." 1>&2
-  echo "Please check Spark and Hadoop configurations, setup gateway roles on this node and try again." 1>&2
-  exit 1
-fi
-
 exec "${CMD[@]}"
-- 
1.7.9.5

